src/runtime/HalideRuntimeHexagonHost.h

```diff
@@ -55,6 +55,11 @@ extern int halide_hexagon_power_hvx_off(void *user_context);
 extern void halide_hexagon_power_hvx_off_as_destructor(void *user_context, void * /* obj */);
 // @}
 
+/** Reserve memory for the Hexagon heap. This can avoid a significant
+ * performance penalty for malloc calls that cause the Hexagon heap to
+ * grow. */
```

Comment:
I went to go try and understand for myself exactly what these are, and it turns out that I think I misunderstood what this does. It doesn't reserve memory, it just changes the strategy for growing the heap. I'm going to close this PR and do something different (simpler from the end user's perspective).